vyper/builtins/functions.py

```diff
@@ -229,6 +229,17 @@ def build_IR(self, expr, context):
 ADHOC_SLICE_NODE_MACROS = ["~calldata", "~selfcode", "~extcode"]
 
 
+# make sure we don't overrun the source buffer, checking for
+# overflow:
+# valid inputs satisfy: `assert start+length <= src_len && start+length > start`
+def _make_slice_bounds_check(start, length, src_len):
+    return [
+        "with",
+        "end",
+        ["add", start, length],
+        ["assert", ["iszero", ["or", ["gt", "end", src_len], ["lt", "end", start]]]],
```

Comment:
yeah, `slice` has inconsistent behavior w.r.t. `length` - statically `length==0` is forbidden, dynamically, it isn't.

i left the comment there until we'd resolve ^

i don't understand the part
>  at least

IMO if overflow happens, then the result must be less than `start`, no?

Reply:
Makes sens!, yeah `at least` was there because I did not think about other potential issue with this at the time of writing but as you say, should be fine as the result would always be smaller than `start`.

Reply:
ah, the comment is wrong, will fix